docs/api.md

```diff
@@ -118,8 +121,46 @@ type AddonsSpec struct {
 }
 
 // TODO maybe we have profiles for scaling behaviors
-type AutoScalerAddonSpec struct {
-    ResourceLimits LimitRange
+type ClusterAutoscaling struct {
+	// Maximum number of nodes in all node groups.
+	// Cluster autoscaler will not grow the cluster beyond this number.
+	// +kubebuilder:validation:Minimum=0
+	MaxNodesTotal int32 `json:"maxNodesTotal"`
+
+	// Should CA scale down the cluster
+    // default: true
+	ScaleDownEnabled bool `json:"enabled,omitempty"`
+
+	// Gives pods graceful termination time before scaling down
+    // default: 600 seconds
+	MaxPodGracePeriod *int32 `json:"maxPodGracePeriod,omitempty"`
+
+	// Maximum time CA waits for node to be provisioned
+    // default: 15 minutes
+	// +kubebuilder:validation:Pattern=^([0-9]+(\.[0-9]+)?(ns|us|µs|ms|s|m|h))+$
+	MaxNodeProvisionTime string `json:"maxNodeProvisionTime,omitempty"`
+
+	// To allow users to schedule "best-effort" pods, which shouldn't trigger
+	// Cluster Autoscaler actions, but only run when there are spare resources available,
+    // default: -10
+	// More info: https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/FAQ.md#how-does-cluster-autoscaler-work-with-pod-priority-and-preemption
+	PodPriorityThreshold *int32 `json:"podPriorityThreshold,omitempty"`
+
+	// BalanceSimilarNodeGroups enables/disables the
+	// `--balance-similar-node-groups` cluster-autocaler feature.
+	// This feature will automatically identify node groups with
+	// the same instance type and the same set of labels and try
+	// to keep the respective sizes of those node groups balanced.
+    // default: false
+	BalanceSimilarNodeGroups *bool `json:"balanceSimilarNodeGroups,omitempty"`
+
+	// Enables/Disables `--ignore-daemonsets-utilization` CA feature flag. Should CA ignore DaemonSet pods when calculating resource utilization for scaling down.
+    // default: false
+	IgnoreDaemonsetsUtilization *bool `json:"ignoreDaemonsetsUtilization,omitempty"`
+
+	// Enables/Disables `--skip-nodes-with-local-storage` CA feature flag. If true cluster autoscaler will never delete nodes with pods with local storage, e.g. EmptyDir or HostPath.
+    // default: true
+	SkipNodesWithLocalStorage *bool `json:"skipNodesWithLocalStorage,omitempty"`
```

Comment:
Maybe controversial as this would change the name, but, what about flipping the name of this to something like `ScaleDownNodesWithLocalStorage` which would default false, then we don't need a true default here and it is, IMO, more obvious to the end user what is going on when this feature is enabled/disabled

Reply:
i'm ok with changing the name if folks think it makes better sense, i have a feeling this one was just copied straight over from the command line flag (`--skip-nodes-with-local-storage`).

Reply:
As with the other bools, I propose removing this field for now and splitting out a separate discussion about a policy enum or union type